src/mlpack/tests/cne_test.cpp

```diff
@@ -0,0 +1,215 @@
+/**
+ * @file cne_test.cpp
+ * @author Marcus Edel
+ * @author Kartik Nighania
+ *
+ * Test file for CNE (Conventional Neural Evolution)
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+
+#include <mlpack/methods/ann/layer/layer.hpp>
+#include <mlpack/methods/ann/ffn.hpp>
+
+#include <mlpack/methods/logistic_regression/logistic_regression.hpp>
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+#include <mlpack/core/optimizers/cne/cne.hpp>
+
+using namespace mlpack;
+using namespace mlpack::ann;
+using namespace mlpack::optimization;
+
+using namespace mlpack::distribution;
+using namespace mlpack::regression;
+
+BOOST_AUTO_TEST_SUITE(CNE_Test);
+
+/**
+ * Training a vanilla network for 2 input XOR function
+ */
+BOOST_AUTO_TEST_CASE(CNE_XOR_TEST)
```

Comment:
We should remove the underscore here, to be consistent with the rest of the codebase.